api/api.go

```diff
@@ -178,38 +178,38 @@ func (s *authedRequester) Do(r *Request) *Response {
 		theurl.RawQuery = encodedParams
 	}
 	req, err := http.NewRequest(reqMethod, theurl.String(), bytes.NewReader(r.Body.Bytes()))
+	if err != nil {
```

Comment:
I presume the old behaviour of setting the `Content-Type` header before bailing for the error was a bug?

Reply:
I presume it would have been, I would imagine that `NewRequest` would return `(nil, error)` and thus we'd try to access `Header` on a `nil` `req`.